glfs.c

```diff
@@ -510,15 +515,50 @@ static int tcmu_glfs_open(struct tcmu_device *dev)
 		goto unref;
 	}
 
-	return 0;
+	ret = glfs_lstat(gfsp->fs, gfsp->hosts->path, &st);
+	if (ret) {
+		tcmu_dev_warn(dev, "glfs_lstat failed: %m\n");
+		goto close;
+	}
+
+	dev_size = tcmu_get_dev_num_lbas(dev) * tcmu_get_dev_block_size(dev);
+	if (st.st_size != dev_size) {
+		if (!tcmu_dev_flags_test(dev, TCMU_DEV_FLAG_REOPENING)) {
```

Comment:
We can just add a new reopen arg to the open callout. I think rbd will want something similar so it is going to be common.

Reply:
But not sure whether there has some exist handlers are using this, like the qemu-tcmu.

Reply:
Will this be a problem ? I it's okay I will update this.

Reply:
We are already not worrying about libtcmu/callout use and non tcmu-runner users. The lib was not being exported anymore.

For the next GPL friendly version of libtcmu we will handle these types of things.

Reply:
Cool, I will update it.